### ISSUE TITLE
Traitlets DeprecationWarnings are generated when starting consoles

### ISSUE THREAD

**Issue by CAM-Gerlach** [spyder-ide/spyder#8174]
## Problem Description

@ccordoba12 Thoughts?

When e.g. running our tests, for any test where a console was open, a flood of the following ``DeprecationWarnings`` were generated, e.g.

```python-traceback
spyder/plugins/tests/test_ipythonconsole.py::test_values_dbg <--- EXAMPLE
  C:\Anaconda3\lib\site-packages\traitlets\config\configurable.py:73: DeprecationWarning: Passing unrecoginized arguments to super(ShellWidget).__init__(local_kernel=True).
  object.__init__() takes no parameters
  This is deprecated in traitlets 4.2.This error will be raised in a future release of traitlets.
    super(Configurable, self).__init__(**kwargs)
```

This is due to [this line](https://github.com/spyder-ide/spyder/blob/a3af9c758e2e9fc25564b73a114e491deb299e11/spyder/plugins/ipythonconsole/widgets/client.py#L151) in client.py that includes ``local_kernel`` as a parameter, which eventually gets passed (apparently via QtConsole) to ``traitlets`` which doesn't like it as it now checks for unrecognized parameters, and an upcoming version will raise an error (thus presumably causing our kernels to never be able to start).

Simply removing it fixes all the warnings, but ``QtConsole`` apparently still uses it upstream of ``Traitlets``, which is presumably why most tests run fine but a few console-related ones in ``test_ipythonconsole.py`` (e.g. ``test_restart_kernel``, ``test_rename_for_slaves``) cause a hard Spyder (Python) crash.

Perhaps a solution (total guess here, I'm clueless about how this infrastructure really works) in the [relevant part](https://github.com/jupyter/qtconsole/blob/f65369c602a659ad0e3bb6e87ccc4df50ba2907c/qtconsole/frontend_widget.py#L169) of QtConsole's ``frontend_widget.py`` would be to strip it off ``kw`` before passing to ``super()`` or just declare it as a separate named parameter in the ``__init__`` signature? Or is there a Spyder level solution to this? I'm uncertain how to best handle the apparent issue, if I'm even understanding it correctly at all.

## Versions

* Spyder version: ``3.x`` and ``master``
* Python version: 3.6.6
* Qt version: 5.9.6
* PyQt version: 5.9.2
* Operating System name/version: Win 8.1

QtConsole 4.4.1 and 4.4.2 (latest)
Traitlets 4.2, 4.3.1 and 4.3.2 (latest)

**Comment by ccordoba12** [comment]
Thanks for tracking the source of these warnings. They are very annoying!!

Please add `local_kernel` here:

https://github.com/spyder-ide/spyder/blob/a3af9c758e2e9fc25564b73a114e491deb299e11/spyder/plugins/ipythonconsole/widgets/shell.py#L66

That should fix this problem.

**Comment by ccordoba12** [comment]
You can do it in PR #8173.

**Comment by ccordoba12** [comment]
Sorry, that won't do the job. It needs to be solved in Qtconsole, as you said, but I'm also unsure how to do it.

**Comment by CAM-Gerlach** [comment]
@ccordoba12 I could try doing the naive solution I suggested (i.e. essentially what you suggest above, except in QtConsole) and see if our tests pass with that version of QtConsole and don't have the deprecation warning. If that's the case, is there anything that needs to be done in Spyder?

**Comment by CAM-Gerlach** [comment]
Okay, looks like that indeed fixed it! Running the full Spyder test suite (including ``--run-slow``) all passes and doesn't emit the warnings with the QtConsole patch applied (running the QtConsole nosetests in my environment gives ``...........Fatal Python error: failed to get random numbers to initialize Python`` whether or not the patch is applied, although QtConsole itself works fine). I'll clean up a few other ``DeprecationWarnings`` and submit the PR to QtConsole. Thanks!

**Comment by CAM-Gerlach** [comment]
@ccordoba12 Done!

**Comment by CAM-Gerlach** [comment]
Fixed in jupyter/qtconsole#313
